proof/crefine/AARCH64/ADT_C.thy

```diff
@@ -855,36 +883,78 @@ lemma carch_tcb_relation_imp_eq:
   apply (case_tac vcpuptr2 ; simp)
   done
 
+lemma tcb_ptr_to_ctcb_ptr_inj:
+  "tcb_ptr_to_ctcb_ptr x = tcb_ptr_to_ctcb_ptr y \<Longrightarrow> x = y"
+  by (auto simp: tcb_ptr_to_ctcb_ptr_def ctcb_offset_def)
+
+lemma
+  shows tcb_at'_tcbBoundNotification:
+  "\<lbrakk>pspace_aligned' as; pspace_distinct' as; valid_tcb' atcb as\<rbrakk>
+   \<Longrightarrow> bound (tcbBoundNotification atcb) \<longrightarrow> ntfn_at' (the (tcbBoundNotification atcb)) as"
+  and tcb_at'_tcbSchedPrev:
+  "\<lbrakk>pspace_aligned' as; pspace_distinct' as; valid_tcb' atcb as\<rbrakk>
+   \<Longrightarrow> tcbSchedPrev atcb \<noteq> None \<longrightarrow> tcb_at' (the (tcbSchedPrev atcb)) as"
+  and tcb_at'_tcbSchedNext:
+  "\<lbrakk>pspace_aligned' as; pspace_distinct' as; valid_tcb' atcb as\<rbrakk>
+   \<Longrightarrow> tcbSchedNext atcb \<noteq> None \<longrightarrow> tcb_at' (the (tcbSchedNext atcb)) as"
```

Comment:
Since all three of these have identical assumptions, would it make sense to compress them via `assumes <assumptions> shows conclusion1 and conclusion2 and conclusion3`?